### ISSUE TITLE
Improve the  "Create a hole in the selected sketch" task (remember previous task settings)

### ISSUE THREAD

**Issue by FreeCAD-Bug-Importer** [FreeCAD/FreeCAD#5875]
Issue imported from https://tracker.freecad.org/view.php?id=3799

* **Reporter:** markus51
* **Date submitted:** 1/26/2019
* **FreeCAD version:** 
* **Category:** Feature
* **Status:** feedback
* **Tags:** #pending-forum

# Original report text

When you have a lot of screws on different planes, it is nerved to refill the form on the "Create a hole in the selected sketch" task.

It will be nice when the task remebers his settings from the last call.

# Steps to reproduce

1.) Draw a new sketch with a circle
2.) Goto "Part Design"
3.) Selcet the new sketch in the tree window
4.) Press "Create a hole in the selected sketch" in the toolbar
5.) The task contains not the last settings, but there are the init values.

# Other bug information

* **Priority:** low
* **Severity:** feature
* **Category:** Feature
* **OS: Win7 **
* **Updated:** 2/6/2021

# Discussion from Mantis ticket

### Comment by Kunda1 2019-01-29 18:12
Why are you still posting 0.17 tickets?
Where is the forum post discussing this with the community?
Please follow the tracker guidelines. Thanks you for your comprehension

---

### Comment by markus51 2019-01-31 10:51
The same is on version 0.18dev ... blabla 

I don't discuss bug's and such basically unusable gui parts, with the community.
It is wasted time. And freecad has a lot of such ugly parts. You can see it on
the first look.

Kunda1@: you shoult be thankful that anyone helps to improve this software.
If you don't want this help --> say it loud ---> then good by

---

### Comment by Kunda1 2019-01-31 14:33
markus51 I am thankful for people contributing and volunteering. There is a way each open source project receives help and in FreeCAD we ask people to do it in a specific way (the large shouting yellow banner at the top of the tracker). If you don't think you apply to that standard you are free to think that, But know that you will receive feedback and orientation from volunteers who need to triage your good will. Also, if you think the interface is ugly you are free to express that and you should know that is very subjective and not constructive. It also won't earn you any extra love from the community. You don't like something about FC then do something about it or deal with it. It's very simple. You are free to become part of the community and follow the guidelines or not (and be responded to accordingly) Thank you for your comprehension.

---

### Comment by Kunda1 2019-04-06 19:17
chrisb can you weigh in on this ticket?

---

### Comment by chrisb 2019-04-09 09:03
For one this is clearly a feature request.

For two OPs attitude is not helpful at all (to avoid the word "unacceptable"). The process has turned out to be sensible. It could possibly be improved, but again this should be discussed in the forum.

For three the basic idea is sensible. Nevertheless it should be discussed in the forum. I can think immediately of different solutions, like the possibility to save current settings, have a list of them, etc. And it might have impact on other dialogs too.

---

### Comment by Kunda1 2020-01-06 16:04
Created https://forum.freecadweb.org/viewtopic.php?f=8&t=42231 for discussion

**Comment by luzpaz** [comment]
@alfrix is still relevant ?

**Comment by alfrix** [comment]
> [@alfrix](https://github.com/alfrix) is still relevant ?

the forum doesn't work at the moment so i cannot check what has been said

you could argue that all the taskpanels last settings should be restored not only the hole one in particular, so this is more of a general feature request of the taskpanels that should be saving and restoring the last state for the document or session

imho in general i wouldn't want it and i would not implement it, the reason is simple, what if you changed some default and then forgot that you did so? now suddenly all your next features will have that setting changed because you didn't noticed that it was changed, and that could cause hours of fixing up in large projects, also this would be a major departure in the current ux workflow, because now you would need to reset the panels if you didn't care about the previous feature values.
TLDR i vote against this, i'll check the forum out of curiosity if it comes back online.
